app/src/main/java/com/github/gotify/login/LoginActivity.java

```diff
@@ -211,12 +228,22 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
                     throw new IllegalArgumentException("file path was invalid");
                 }
 
-                String content = Utils.readFileFromStream(fileStream);
-                String name = getNameOfCertContent(content);
-
-                // temporarily set the contents (don't store to settings until they decide to login)
-                caCertContents = content;
-                advancedDialog.showRemoveCACertificate(name);
+                if (requestCode == CA_FILE_SELECT_CODE) {
+                    String content = Utils.readFileFromStream(fileStream);
+                    String name = getNameOfCertContent(content);
+                    // temporarily set the contents (don't store to settings until they decide to login)
+                    caCertContents = content;
+                    advancedDialog.showRemoveCACertificate(name);
+                } else if (requestCode ==  CLI_CERT_FILE_SELECT_CODE) {
+                    String content = Utils.binaryFileToBase64(fileStream);
+
+                    clientCertContents = content;
+                    String path = uri.getLastPathSegment();
+
+                    settings.clientCert(clientCertContents);
+                    settings.clientCertUri(path);
```

Comment:
Why set the settings here? Now you have clientCertContents and settings.clientCert which have the same value. Values should only be set on the settings inside com.github.gotify.login.LoginActivity#onCreatedClient

Also the settings.clientCert won't be unset on remove client certificate, so the setting/variable become out of sync.
```
.onClickRemoveClientCertificate(() -> {
    clientCertContents = null;
    settings.clientCertPass("");
    clientCertPassword = "";
    settings.clientCertUri(null);
    clientCertUri = null;
})
```

Reply:
Resetting the setting values in that handler is done because I thought it was similar to the existing `onClickRemoveCaCertificate` above reset the ca cert contents on removal. 

I will add the missing clientCert reset.